src/frontend/screens/WineManager/components/WineItem/index.tsx

```diff
@@ -33,20 +33,26 @@ const WineItem = ({
     progress: ProgressInfo
   }>({ state: 'idle', progress: { percentage: 0, avgSpeed: 0, eta: Infinity } })
 
-  if (version) {
-    window.api.handleProgressOfWineManager(
-      version,
-      (
-        e: Electron.IpcRendererEvent,
-        progress: {
-          state: State
-          progress: ProgressInfo
-        }
-      ) => {
-        setProgress(progress)
-      }
-    )
-  }
+  useEffect(() => {
+    if (version) {
+      const removeWineManagerDownloadListener =
+        window.api.handleProgressOfWineManager(
+          version,
+          (
+            e: Electron.IpcRendererEvent,
+            progress: {
+              state: State
+              progress: ProgressInfo
+            }
+          ) => {
+            setProgress(progress)
+          }
+        )
+      return removeWineManagerDownloadListener
+    }
+    /* eslint-disable @typescript-eslint/no-empty-function */
+    return () => {}
+  }, [])
```

Comment:
Is something like this not possible?
The api need to have a remove function:
```typescript
export const removeProgressOfWineManager = (
  version: string,
  callback: (
    e: Electron.IpcRendererEvent,
    progress: {
      state: State
      progress: ProgressInfo
    }
  ) => void
) => {
    ipcRenderer.removeListener('progressOfWineManager' + version, callback)
}
```

And we just call the remover funcion inside the unmount of useEffect.
```suggestion
  useEffect(() => {
    const onProgress = (e: Electron.IpcRendererEvent, progress: {state: State, progress: ProgressInfo) 
      => {setProgress(progress)}
    if (version) {
        window.api.handleProgressOfWineManager(
          version,
          onProgress
        )
    }
    
    // unmount
    return () => {
        window.api.removeProgressOfWineManager(version, onProgress)
    }
  }, [])
```

Reply:
I tried this recently and the remove listener wasn't working so on every render it would just subscribe again and then I would get n messages from the back where n is the number of component renders. Here's an issue that describes the same problem and returning the ()=>removeListener callback to the frontend works as a solution https://github.com/reZach/secure-electron-template/issues/43#issuecomment-772303787. This also generally fits the pattern of listening/removing listener in the same context and is easier than a separate call imo